### PR TITLE
[MCU-2756] JWT Sample

### PR DESCRIPTION
**Pull request by ConorOD**
## Changes
* Added simple JWT sample app

### Reviewers
@zendesk/adventure-android @zendesk/two-brothers-android @zendesk/atom-android @baz8080

### CODE REVIEW

File: chat_v2_jwt_sample/src/main/java/com/zendesk/chat_v2/jwt/sample/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        btnStart = findViewById(R.id.btnStart);
+
+        btnSet.setOnClickListener(clickListener);
+        btnReset.setOnClickListener(clickListener);
+        btnStart.setOnClickListener(clickListener);
+
+        jwtAuthenticator = new JwtAuthenticator() {
+            @Override
+            public void getToken(JwtCompletion jwtCompletion) {
+                //Fetch or generate the JWT token at this point
```
Comment (bridgeri127) [inline chat_v2_jwt_sample/src/main/java/com/zendesk/chat_v2/jwt/sample/MainActivity.java:69]: Maye be should lay this out a little differently the indentation is strange

**Review by bridgeri127 — COMMENTED** [review]
Looks good, few small comments

File: chat_v2_jwt_sample/src/main/java/com/zendesk/chat_v2/jwt/sample/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+public class MainActivity extends AppCompatActivity {
+
+    AppCompatButton btnSet;
+    AppCompatButton btnReset;
+    AppCompatButton btnStart;
+
+    JwtAuthenticator jwtAuthenticator;
+
+    View.OnClickListener clickListener = new View.OnClickListener() {
```
[the remark this answers is not in the document]
Comment (ConorOD) [reply chat_v2_jwt_sample/src/main/java/com/zendesk/chat_v2/jwt/sample/MainActivity.java:26]: Fair I did that at first and changed it. Tried to be cool and failed :(

File: chat_v2_jwt_sample/src/main/java/com/zendesk/chat_v2/jwt/sample/Global.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        // Enable logging
+        Logger.setLoggable(true);
+
+        if (StringUtils.isEmpty(CHAT_ACCOUNT_KEY)) {
+            missingCredentials = true;
```
[the remark this answers is not in the document]
Comment (bridgeri127) [reply chat_v2_jwt_sample/src/main/java/com/zendesk/chat_v2/jwt/sample/Global.java:24]:
```suggestion
            missingCredentials = true;
            return;
```

**Review by bridgeri127 — APPROVED** [review]

**Pull request merged**